gustaf/io/hmascii.py

```diff
@@ -0,0 +1,313 @@
+"""gustaf/gustaf/io/hmascii.py
+
+Read a mesh and boundaries from a HyperMesh .hmascii file.
+Output is not supported.
+
+This implementation requires a certain structure of the model. It only imports
+volume elements of a single type, which is either tetrahedra or hexahedra.
+They all must be in a single component (default name: 'volume').
+Boundaries can be imported as vertex groups, which can be converted to face
+groups. To enable the creation of a vertex group, a component must be created
+in HyperMesh. This component must contain surface elements of the current
+subelement type. They must use the same node IDs as the volume elements and
+not just share their coordinates.
+An initial surface component can be created in HyperMesh using:
+tools -> faces
+"""
+
+import logging
+import numpy as np
+
+from gustaf.faces import Faces
+from gustaf.volumes import Volumes
+from gustaf.edges import Edges
+
+
+class HMLine:
+    """Parse a line from an HMASCII file.
+    """
+
+    __slots__ = ["name", "values"]
+
+    def __init__(self, line):
+        """
+        Parameters
+        -----------
+        line: str
+
+        Returns
+        --------
+        None
+        """
+        self.name = ''
+        self.values = ''
+
+        if line[0] == '*':
+            parts = line[1:].split('(')
+            self.name = parts[0]
+            if len(parts) > 1:
+                self.values = [
+                        value.strip('"') for value in parts[1][:-2].split(',')
+                ]
+
+
+class HMElementType:
+    """Store HyperMesh element type information.
+    """
+
+    __slots__ = ["number_of_nodes", "subelement", "MeshType"]
+
+    def __init__(self, number_of_nodes, MeshType, subelement=''):
+        """
+        Parameters
+        -----------
+        number_of_nodes: int
+        subelement: str
+        MeshType: Mesh
+
+        Returns
+        --------
+        None
+        """
+        self.number_of_nodes = int(number_of_nodes)
+        self.MeshType = MeshType
```

Comment:
could you rename the `MeshType` variable name to be more inline with other variable names?